test/SILOptimizer/cast_folding_conditional_conformance.swift

```diff
@@ -0,0 +1,18 @@
+// RUN: %target-swift-frontend %s -emit-sil -O -o - | %FileCheck %s
+// RUN: %target-swift-frontend %s -emit-ir -o /dev/null
+
+// rdar://problem/38694450
+
+protocol P {}
+extension Array: P where Element: P {}
+struct X {}
+// CHECK: unconditional_checked_cast_addr Array<X> in {{%[0-9]*}} : $*Array<X> to P in {{%[0-9]*}} : $*P
```

Comment:
Put this in a function and use a CHECK-LABEL: to match the function declaration line